src/pkg/cli/getServices.go

```diff
@@ -62,5 +70,44 @@ func GetServices(ctx context.Context, projectName string, provider client.Provid
 		servicesResponse.Services[i] = nil
 	}
 
-	return term.Table(printServices, "Service", "Deployment", "PublicFqdn", "PrivateFqdn", "Status")
+	return term.Table(printServices, "Service", "Deployment", "PublicFqdn", "PrivateFqdn", "State")
+}
+
+func UpdateServiceStates(ctx context.Context, serviceInfos []*defangv1.ServiceInfo) {
+	// Create a context with a timeout for HTTP requests
+	ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
+	defer cancel()
+	var wg sync.WaitGroup
+
+	for _, serviceInfo := range serviceInfos {
+		for _, endpoint := range serviceInfo.Endpoints {
+			if !strings.Contains(endpoint, ":") {
+				wg.Add(1)
+				go func(serviceInfo *defangv1.ServiceInfo) {
+					defer wg.Done()
+					url := "https://" + endpoint + "/" // TODO: use serviceInfo.Healthcheck
+					// Use the regular net/http package to make the request without retries
+					req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+					if err != nil {
+						term.Errorf("Failed to create healthcheck request for %q at %s: %s", serviceInfo.Service.Name, url, err.Error())
```

Comment:
Should not be printing in Go routines, but for debug is fine I suppose.
```suggestion
						term.Debugf("Failed to create healthcheck request for %q at %s: %s", serviceInfo.Service.Name, url, err.Error())
```

Reply:
I wouldn't. debug is fine.

Reply:
yeah, so I'm ok with debug prints, but def no error/warn/info/print calls.